source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -533,6 +534,32 @@ bool Filter::parseTlvs(const uint8_t* buf, size_t len) {
     absl::string_view tlv_value(reinterpret_cast<char const*>(buf + idx), tlv_value_length);
     auto key_value_pair = config_->isTlvTypeNeeded(tlv_type);
     if (nullptr != key_value_pair) {
+      if (Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.use_typed_metadata_in_proxy_protocol_listener")) {
+        auto& typed_filter_metadata = (*cb_->dynamicMetadata().mutable_typed_filter_metadata());
+        std::string metadata_key = key_value_pair->metadata_namespace().empty()
+                                       ? "envoy.filters.listener.proxy_protocol"
+                                       : key_value_pair->metadata_namespace();
+        const auto typed_proxy_filter_metadata = typed_filter_metadata.find(metadata_key);
+        envoy::extensions::filters::listener::proxy_protocol::v3::TlvsMetadata tlvs_metadata;
+        auto status = absl::OkStatus();
+        if (typed_proxy_filter_metadata != typed_filter_metadata.end()) {
+          status = MessageUtil::unpackTo(typed_proxy_filter_metadata->second, tlvs_metadata);
+        }
+        if (!status.ok()) {
+          ENVOY_LOG(warn, "proxy_protocol: Failed to unpack typed metadata for TLV type ",
```

Comment:
Maybe `ENVOY_LOG_PERIODIC` in case this starts happening on every connection for some reason